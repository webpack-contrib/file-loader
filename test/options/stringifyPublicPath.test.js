/* eslint-disable
  prefer-destructuring,
*/
import webpack from '../helpers/compiler';

describe('Options', () => {
  describe('stringifyPublicPath', () => {
    describe('{Boolean}', () => {
      test('True (Default)', async () => {
        const config = {
          loader: {
            test: /(png|jpg|svg)/,
            options: {
              publicPath: url => `some/path/${url}`,
              stringifyPublicPath: true,
            },
          },
        };

        const stats = await webpack('fixture.js', config);
        const { assets, source } = stats.toJson().modules[1];

        expect({ assets, source }).toMatchSnapshot();
      });

      test('False', async () => {
        const config = {
          loader: {
            test: /(png|jpg|svg)/,
            options: {
              publicPath: url => `(window.CDN_PATH ? window.CDN_PATH + "${url}" : "${url}")`,
              stringifyPublicPath: false,
            },
          },
        };

        const stats = await webpack('fixture.js', config);
        const { assets, source } = stats.toJson().modules[1];

        expect({ assets, source }).toMatchSnapshot();
      });
    });
  });
});
