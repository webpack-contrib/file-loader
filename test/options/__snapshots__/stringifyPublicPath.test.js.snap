// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options stringifyPublicPath {Boolean} False 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = (window.CDN_PATH ? window.CDN_PATH + \\"9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\" : \\"9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\");",
}
`;

exports[`Options stringifyPublicPath {Boolean} True (Default) 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = \\"some/path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;
